common/utils/src/main/java/org/apache/spark/internal/Logger.java

```diff
@@ -0,0 +1,184 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.internal;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+
+import org.apache.logging.log4j.CloseableThreadContext;
+import org.apache.logging.log4j.message.MessageFactory;
+import org.apache.logging.log4j.message.ParameterizedMessageFactory;
+
+public class Logger {
+
+  private static final MessageFactory MESSAGE_FACTORY = ParameterizedMessageFactory.INSTANCE;
+  private final org.slf4j.Logger slf4jLogger;
+
+  Logger(org.slf4j.Logger slf4jLogger) {
+    this.slf4jLogger = slf4jLogger;
+  }
+
+  public void error(String msg) {
+    slf4jLogger.error(msg);
+  }
+
+  public void error(String msg, Throwable throwable) {
+    slf4jLogger.error(msg, throwable);
+  }
+
+  public void error(String msg, MDC... mdcs) {
+    if (mdcs == null || mdcs.length == 0) {
+      slf4jLogger.error(msg);
+    } else if (slf4jLogger.isErrorEnabled()) {
+      withLogContext(msg, mdcs, null, mt -> slf4jLogger.error(mt.message));
+    }
+  }
+
+  public void error(String msg, Throwable throwable, MDC... mdcs) {
+    if (mdcs == null || mdcs.length == 0) {
+      slf4jLogger.error(msg, throwable);
+    } else if (slf4jLogger.isErrorEnabled()) {
+      withLogContext(msg, mdcs, throwable, mt -> slf4jLogger.error(mt.message, mt.throwable));
+    }
+  }
+
+  public void warn(String msg) {
+    slf4jLogger.warn(msg);
+  }
+
+  public void warn(String msg, Throwable throwable) {
+    slf4jLogger.warn(msg, throwable);
+  }
+
+  public void warn(String msg, MDC... mdcs) {
+    if (mdcs == null || mdcs.length == 0) {
+      slf4jLogger.warn(msg);
+    } else if (slf4jLogger.isWarnEnabled()) {
+      withLogContext(msg, mdcs, null, mt -> slf4jLogger.warn(mt.message));
+    }
+  }
+
+  public void warn(String msg, Throwable throwable, MDC... mdcs) {
+    if (mdcs == null || mdcs.length == 0) {
+      slf4jLogger.warn(msg);
+    } else if (slf4jLogger.isWarnEnabled()) {
+      withLogContext(msg, mdcs, throwable, mt -> slf4jLogger.warn(mt.message, mt.throwable));
+    }
+  }
+
+  public void info(String msg) {
+    slf4jLogger.info(msg);
+  }
+
+  public void info(String msg, Throwable throwable) {
+    slf4jLogger.info(msg, throwable);
+  }
+
+  public void info(String msg, MDC... mdcs) {
+    if (mdcs == null || mdcs.length == 0) {
+      slf4jLogger.info(msg);
+    } else if (slf4jLogger.isInfoEnabled()) {
+      withLogContext(msg, mdcs, null, mt -> slf4jLogger.info(mt.message));
+    }
+  }
+
+  public void info(String msg, Throwable throwable, MDC... mdcs) {
+    if (mdcs == null || mdcs.length == 0) {
+      slf4jLogger.info(msg);
```

Comment:
We should have the `throwable` here. Also, we need test case for this.

Reply:
Okay

Reply:
@gengliangwang 
I have submitted a PR to fix it: https://github.com/apache/spark/pull/47095